src/components/DetailsHeader.jsx

```diff
@@ -1,41 +1,49 @@
 import React from 'react';
 import '../styles/DetailsHeader.css';
 import findData from '../services/findData';
+import FavoriteButton from './FavoriteButton';
+import CopyToClipboard from './CopyToClipboard';
 
-function mealHeader(data) {
+function mealHeader(data, id) {
```

Comment:
Nas funções `drinkHeader` e `mealHeader`, percebeu que elas contem a mesma estrutura?

Porque não verifica na URL em qual página ela está e retorna os elementos de acordo com a página? Exemplo:
```suggestion
function mealHeader(data, id, currentURL) {
  if(currentURL.includes('meal'){
    const imageSrc = strMealThumb
    // e por ai vai...
  }
}
```

Se precisar de um exemplo, fiz algo semelhante na página de ingredientes.

Reply:
Excelente ideia!